src/Razor/Microsoft.NET.Sdk.Razor/src/ConcatenateCssFiles.cs

```diff
@@ -13,21 +14,72 @@ namespace Microsoft.AspNetCore.Razor.Tasks
+
+        [Required]
+        public ITaskItem[] ScopedCssFiles { get; set; }
+
+        [Required]
+        public ITaskItem[] ProjectBundles { get; set; }
+
         [Required]
-        public ITaskItem[] FilesToProcess { get; set; }
+        public string ScopedCssBundleBasePath { get; set; }
 
         [Required]
         public string OutputFile { get; set; }
 
         public override bool Execute()
         {
+            if (ProjectBundles.Length > 0)
+            {
+                Array.Sort(ProjectBundles, _fullPathComparer);
+            }
+            Array.Sort(ScopedCssFiles, _fullPathComparer);
+
             var builder = new StringBuilder();
-            var orderedFiles = FilesToProcess.OrderBy(f => f.GetMetadata("FullPath")).ToArray();
-            for (var i = 0; i < orderedFiles.Length; i++)
+            if (ProjectBundles.Length > 0)
             {
-                var current = orderedFiles[i];
-                builder.AppendLine($"/* {current.GetMetadata("BasePath").Replace("\\","/")}{current.GetMetadata("RelativePath").Replace("\\","/")} */");
-                foreach (var line in File.ReadLines(FilesToProcess[i].GetMetadata("FullPath")))
+                // We are importing bundles from other class libraries and packages, in that case we need to compute the
+                // import path relative to the position of where the final bundle will be.
+                // Our final bundle will always be at "<<CurrentBasePath>>/scoped.styles.css"
+                // Other bundles will be at "<<BundleBasePath>>/bundle.bdl.scp.css"
+                // The base and relative paths can be modified by the user, so we do a normalization process to ensure they
+                // are in the shape we expect them before we use them.
+                // We normalize path separators to '\' from '/' which is what we expect on a url. The separator can come as
+                // '\' as a result of user input or another MSBuild path normalization operation. We always want '/' since that
+                // is what is valid on the url.
+                // We remove leading and trailing '/' on all paths to ensure we can combine them properly. Users might specify their
+                // base path with or without forward and trailing slashes and we always need to make sure we combine them appropriately.
+                // These links need to be relative to the final bundle to be independent of the path where the main app is being served.
+                // For example:
+                // An app is served from the "subdir" path base, the main bundle path on disk is "MyApp/scoped.styles.css" and it uses a
+                // library with scoped components that is placed on "_content/library/bundle.bdl.scp.css".
+                // The resulting import would be "import '../_content/library/bundle.bdl.scp.css'".
```

Comment:
I'm unsure about this. I would have expected the CSS import to look like `@import '_content/library/bundle.bdl.scp.css'`, i.e., *not* have a leading `..`. That's because the main app's bundle is served from the content root, not some subdirectory inside it.

Is it just an out-of-date comment, or might there be a functional issue here?

Reply:
@SteveSandersonMS the main bundle is served from the content root only if you don't override the SWA base path, if you do, it is served from there.

Reply:
Ah OK makes sense. Thanks!